src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs

```diff
@@ -19,6 +18,40 @@
 
 namespace Stratis.FederatedPeg.Features.FederationGateway.TargetChain
 {
+    public class StatusChangeTracker:Dictionary<ICrossChainTransfer, CrossChainTransferStatus?>
+    {
+        /// <summary>
+        /// Records changes to transfers for the purpose of synchronizing the transient lookups after the DB commit.
+        /// </summary>
+        /// <param name="transfer">The cross-chain transfer to update.</param>
+        /// <param name="status">The new status.</param>
+        /// <param name="blockHash">The block hash of the partialTranction.</param>
+        /// <param name="blockHeight">The block height of the partialTransaction.</param>
+        public void SetTransferStatus(ICrossChainTransfer transfer, CrossChainTransferStatus? status = null, uint256 blockHash = null, int blockHeight = 0)
```

Comment:
Could there be a default status - like 'New' ? or will null be enough for this ?

Reply:
Ok - maybe your comment could be useful as a remark.

Reply:
Done.